requirements.txt

```diff
@@ -6,8 +6,3 @@ dill>=0.3
 python-dateutil>=2.8.0
 stevedore>=3.0.0
 typing-extensions
-
-# If updating the version range here, consider updating the 
-# list of symengine dependencies used in the cross-version tests
-# in 'test/qpy_compat/run_tests.sh' and 'test/qpy_compat/qpy_test_constraints.txt'
-symengine>=0.11,<0.14
```

Comment:
Since QPY can still rely on `symengine`, do we need to document these constraints somewhere still? E.g. in `requirements-optional` or `constraints`?

Reply:
`requirements-optional.txt` and constraints is actually only for testing and ci, I can add a new optional variant to the package metadata so users can do `pip install qiskit[qpy-compat]` which is how we would document this. I mostly didn't because qpu parsing will raise an exception with the error message about the symengine version if it detects the mismatch already. But it doesn't hurt to add it there

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/14389/commits/dd4a32df746dea033455020b98ca8e95033097eb